MessageCenter3/Models/Message/MessageRepository.cs

```diff
@@ -88,49 +84,48 @@ private void AddNewMessageReipient(int messageId, int recipientId, RecipientType
             }   
         }
 
-        public List<MessageOutputModel> GetMessagesFromTo(int senderId, int recipientId, RecipientType recipientType)
+        public List<MessageOutputModel> GetDialogue(UserMessageRequestModel request)
         {
-            string sqlQuery = string.Empty;
-
-            switch (recipientType)
-            {
-                case RecipientType.User:
-                    sqlQuery = @"
-                        SELECT [User].Id AS CollocutorId, [User].Name AS CollocutorName, 
-                            SenderId, TypeId, CreateDateAndTime, Text, Message.Id AS Id 
+            string sqlQuery = sqlQuery = @"
+                        SELECT RecipTable.RecipientId, [User].Name AS SenderName, 
+                                SenderId, TypeId, CreateDateAndTime, Text, Message.Id AS Id 
                             FROM Message 
 	                            JOIN MessageUserRecipient RecipTable
 	                                ON Message.Id = RecipTable.MessageId
                                 JOIN [User]
-                                    ON RecipTable.RecipientId = [User].Id
-                                WHERE Message.SenderId = @senderId
-                                AND RecipTable.RecipientId = @recipientId";
-                    break;
-                case RecipientType.Group:
-                    sqlQuery = @"
-                        SELECT [User].Id AS CollocutorId, [User].Name AS CollocutorName, 
-                            SenderId, TypeId, CreateDateAndTime, Text, Message.Id AS Id 
+                                    ON Message.SenderId = [User].Id
+                                WHERE Message.SenderId IN (@CurrentUserId, @CollocutorId)
+                                AND RecipTable.RecipientId IN (@CurrentUserId, @CollocutorId)
+                                AND Message.TypeId = @MessageType
+                        ORDER BY Message.CreateDateAndTime
```

Comment:
For the future, you can use CreatedOn instead of CreateDateAndTime

Reply:
Got it. Thanks!